### DISCUSSION TITLE
Flight stability with CrazyFlie while using Loco Position System

### DISCUSSION THREAD

**Discussion by StewenceNewton** [bitcraze/discussions#1190] [Q&A] [answered]
Hi, My name is newton I have been working on these loco positioning systems along with the Crazyflie to achieve swarm and these are responding to the code properly but the problem is with the flight stability. These are not staying in the particular position moving up and down not able to hold in the position, I tried with the TWR, TdoA2 and TdoA3 mode. Can you help on this thing and I have used 8 loco nodes around 3mx5mx2m (XxYxZ) space. While controlling with cfclient also it is not able to stay in the single position and i'm attaching the video reference as a google drive link.

https://drive.google.com/file/d/1EBtB8Gdcpx-bC5-dzYeHgUsll5081CQ4/view?usp=sharing

**Comment by CalvinKoung** [comment]
Maybe you can try to slow down the flight speed ?

**Comment by knmcguire** [comment, marked as answer]
I looked at the video and I don't think there is a drifting issue here. The loco positioning system does have larger inaccuracies than Lighthouse or mocap and even in our own lab, the Crazyflie with the loco positioning system tends to move withing half a meter area. In your flight space I see a lot of objects that can affect the quality even further.

If you want to see if you can improve it, I suggest you take a look at the robust TWR / TDOA mode.  Here is the blogpost of those that implemented it https://www.bitcraze.io/2021/06/robust-kalman-filter-for-ultra-wideband-localization/

**Reply by CalvinKoung** [reply]
Wow, that so amazing. I saw the blog said this functionality can be turned on by setting a parameter (robustTWR or robustTdoa) in estinator_kalman.c so if I want to use this robust Kalman filter that I put the fasle modify true and use client to update the firmware it is correct ?

**Comment by StewenceNewton** [comment]
@knmcguire Thank you for your response I'll follow those instructions and try to implement again.
And is their any chance of interference to this Ultra wide band(LPS) with WiFi or 5G mobile signals.

**Reply by ataffanel** [reply]
In theory, and in our experience, there is not a lot of interference between 5G/wifi and UWB: UWB works on a much wider bandwidth than both Wifi and 5G, and it tends to work under the noise floor, so chances of interference are reduced.

Though, this is something that needs to be tested on your side in your particular environment.

**Reply by StewenceNewton** [reply]
Thank you for your response, I'll check on it

**Comment by StewenceNewton** [comment]
@knmcguire hi good morning, I hope you are fine there, I have an issue with my flapper(nimble) drone the thing is servos get damaged and i want to know the specifications or else can you help in purchasing the exact match for that servos.

**Reply by knmcguire** [reply]
Hi! Could you contact us at contact _at_ bitcraze.io for this? Also this is something different than the original thread was meant for, so 'normally' it would have been better to make a new discussion thread, but since this is hardware specific, please send an email instead.

**Reply by StewenceNewton** [reply]
Thank you for your response
